pxd_bio_blkmq.c

```diff
@@ -256,7 +257,9 @@ static int prep_root_bio(struct fp_root_context *fproot) {
                 return 0;
         }
 
-        rq_for_each_segment(bv, rq, rq_iter) nr_bvec++;
+        if (!specialops)
+                rq_for_each_segment(bv, rq, rq_iter) nr_bvec++;
+
```

Comment:
This is the root cause of the crash. For discard IOs, the above function is indeterminate. Ideally should return zero vectors, but starting from 5.7 kernels discards/write same can carry special payloads. For px virtual device, no additional payload is needed. Compute bvecs only if not a special operation currently only discard.